### PR TITLE
new module to support JDK17 types

### PR DESCRIPTION
**Pull request by hg-ms**
added support for Java 17 implementations of
java.util.ImmutableCollections.Set12<E> and
java.util.ImmutableCollections.List12<E>

### CODE REVIEW

File: docs/modules/storage/pages/addendum/specialized-type-handlers.adoc [hunk 1]
```diff
@@ -111,3 +111,21 @@ If you want a more optimized version of JDK collection handlers, you can utilize
...
+		<groupId>one.microstream</groupId>
+		<artifactId>microstream-persistence-binary-jdk17</artifactId>
+		<version>{maven-version}</version>
+	</dependency>
+</dependencies>
+----
+
+[source, java]
+----
+final EmbeddedStorageFoundation<?> foundation = EmbeddedStorage.Foundation();
```
Comment (fh-ms) [inline docs/modules/storage/pages/addendum/specialized-type-handlers.adoc:129]: This class probably should be renamed:
BinaryHandlersJDK15 -> BinaryHandlersJDK17

**Review by fh-ms — APPROVED** [review]

File: persistence/binary-jdk17/pom.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	</parent>
+
+	<artifactId>microstream-persistence-binary-jdk17</artifactId>
+	
+	<name>MicroStream Persistence JDK11</name>
+	<description>MicroStream Persistence Specialized Type Handlers for JDK17</description>
+	
+	<properties>
+		<maven.compiler.source>11</maven.compiler.source>
+		<maven.compiler.target>11</maven.compiler.target>
```
Comment (zdenek-jonas) [inline persistence/binary-jdk17/pom.xml]: 11?

File: persistence/binary-jdk17/src/main/java/one/microstream/persistence/binary/jdk17/java/util/BinaryHandlerImmutableCollectionsSet12.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package one.microstream.persistence.binary.jdk17.java.util;
+
+/*-
+ * #%L
+ * MicroStream Persistence JDK11
```
Comment (zdenek-jonas) [inline persistence/binary-jdk17/src/main/java/one/microstream/persistence/binary/jdk17/java/util/BinaryHandlerImmutableCollectionsSet12.java]: Please generate license header with correct name again.

File: persistence/binary-jdk17/src/main/java/one/microstream/persistence/binary/jdk17/types/BinaryHandlersJDK17.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package one.microstream.persistence.binary.jdk17.types;
+
+/*-
+ * #%L
+ * microstream-persistence-binary-jdk8
```
Comment (zdenek-jonas) [inline persistence/binary-jdk17/src/main/java/one/microstream/persistence/binary/jdk17/types/BinaryHandlersJDK17.java]: bad name of module

**Review by zdenek-jonas — CHANGES_REQUESTED** [review]

File: persistence/binary-jdk17/pom.xml (hunk 2 above)
Comment (hg-ms) [reply persistence/binary-jdk17/pom.xml]: done

File: persistence/binary-jdk17/src/main/java/one/microstream/persistence/binary/jdk17/java/util/BinaryHandlerImmutableCollectionsSet12.java (hunk 3 above)
Comment (hg-ms) [reply persistence/binary-jdk17/src/main/java/one/microstream/persistence/binary/jdk17/java/util/BinaryHandlerImmutableCollectionsSet12.java]: done

**Review by zdenek-jonas — APPROVED** [review]

**Pull request merged**